internal/machine/machine_set.go

```diff
@@ -127,7 +128,7 @@ func (ms *machineSet) ReleaseLeases(ctx context.Context) error {
 	hadError := false
 	for err := range results {
 		contextTimedOutOrCanceled := errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled)
-		if err != nil && (!contextWasAlreadyCanceled || !contextTimedOutOrCanceled) {
+		if err != nil && (!contextWasAlreadyCanceled || !contextTimedOutOrCanceled || !strings.Contains(err.Error(), "lease not found")) {
```

Comment:
Is the goal to ignore the error if it the lease already expired? if so, I think you have to change the condition because it is making it more narrow than it should.

If you apply De Morgan's law it reads:
```go
if err != nil && !(contextWasAlreadyCanceled && contextTimedOutOrCanceled && strings.Contains(err.Error(), "lease not found")) {
```